module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,61 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <utility>
+
+using vector_of_pairs = std::vector<std::pair<uint8_t, uint8_t>>;
+using array_of_arrays = std::array<std::array<uint8_t, width>, height>;
+
+vector_of_pairs compressGrayscale(const array_of_arrays& bitmap) {
+    vector_of_pairs compressed{};
```

Comment:
Reserve memory, and then use `shrink_to_fit`.

Reply:
Done in latest commit, thanks :)